### PR TITLE
[CAFV-389] support for cluster class in v1beta3

### PR DESCRIPTION
**Pull request by Anirudh9794**
## Description
Please provide a brief description of the changes proposed in this Pull Request

- added type v1beta3/VCDClusterTemplate
- added conversion logic for converting v1beta2/VCDClusterTemplate to v1beta3/VCDClusterTemplate
- added defaulting and validation webhooks. We do not expect any changes to be done for the VCDClusterTemplate API object once it is created. Hence the ValidationWebhook will deny any update to the API object.

## Checklist
- [x] tested locally
- [x] updated any relevant dependencies
- [x] updated any relevant documentation or examples - updated template/cluster-class-template.yaml

## API Changes
Are there API changes?
- [x] Yes
- [ ] No

If yes, please fill in the below

1. Updated conversions?
    - [x] Yes
    - [ ] No
    - [ ] N/A
2. Updated CRDs?
    - [x] Yes
    - [ ] No
    - [ ] N/A
3. Updated infrastructure-components.yaml?
    - [x] Yes
    - [ ] No
    - [ ] N/A
4. Updated `./examples/capi-quickstart.yaml`?
    - [ ] Yes
    - [ ] No
    - [x] N/A - since this is a clusterclass implementation and capi-quickstart doesnt use clusterclass
5. Updated necessary files under `./infrastructure-vcd/v1.0.0/`?
   - [x] Yes
   - [ ] No
   - [ ] N/A

## Issue
If applicable, please reference the relevant issue

Fixes #

- - -
This change is [Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/604)

### CODE REVIEW

File: api/v1beta3/vcdcluster_webhook.go [hunk 1]
```diff
@@ -35,7 +35,8 @@ func (r *VCDCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
 
 // TODO(user): EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 
-// +kubebuilder:webhook:path=/mutate-infrastructure-cluster-x-k8s-io-v1beta3-vcdcluster,mutating=true,failurePolicy=fail,sideEffects=None,groups=infrastructure.cluster.x-k8s.io,resources=vcdclusters,verbs=create;update,versions=v1beta3,name=mutation.vcdcluster.infrastructure.cluster.x-k8s.io,admissionReviewVersions=v1
+//+kubebuilder:webhook:path=/mutate-infrastructure-cluster-x-k8s-io-v1beta3-vcdcluster,mutating=true,failurePolicy=fail,sideEffects=None,groups=infrastructure.cluster.x-k8s.io,resources=vcdclusters,verbs=create;update,versions=v1beta3,name=mutation.vcdcluster.kb.io.infrastructure.cluster.x-k8s.io,admissionReviewVersions=v1
```
Comment (arunmk) [inline api/v1beta3/vcdcluster_webhook.go:38]: The path of the webhook doesn't need to be changed

**Review by arunmk — APPROVED** [review]
LGTM. There is one webhook whose GVK has changed. We should revert that one.

File: api/v1beta3/vcdclustertemplate_webhook.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *VCDClusterTemplate) Default() {
+	vcdclustertemplatelog.Info("default", "name", r.Name)
+
+	// TODO(user): fill in your defaulting logic.
+}
+
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
+//+kubebuilder:webhook:path=/validate-infrastructure-cluster-x-k8s-io-v1beta3-vcdclustertemplate,mutating=false,failurePolicy=fail,sideEffects=None,groups=infrastructure.cluster.x-k8s.io,resources=vcdclustertemplates,verbs=create;update,versions=v1beta3,name=validation.vcdclustertemplate.infrastructure.cluster.x-k8s.io,admissionReviewVersions=v1
```
Comment (viveksyngh) [inline api/v1beta3/vcdclustertemplate_webhook.go:53]: nit: If there is no validation for create then we can remove from the webhook configuration

**Review by viveksyngh — APPROVED** [review]

**Review by sahithi — APPROVED** [review]

File: api/v1beta3/vcdclustertemplate_webhook.go (hunk 2 above)
Comment (Anirudh9794) [reply api/v1beta3/vcdclustertemplate_webhook.go:53]: Discussed about this. We may want to add some logic in the future. So I will keep this as is.

**Pull request merged**